ironfish/src/rpc/routes/wallet/createAccount.ts

```diff
@@ -45,10 +59,23 @@ routes.register<typeof CreateAccountRequestSchema, CreateAccountResponse>(
   CreateAccountRequestSchema,
   async (request, context): Promise<void> => {
     AssertHasRpcContext(request, context, 'wallet')
+    if (
+      request.data.createdAt?.hash === undefined ||
+      request.data.createdAt?.sequence === undefined
+    ) {
+      request.data.createdAt = undefined
+    }
```

Comment:
You can delete this entire block of code according to your spec above.

Reply:
The yup spec? Yup doesn't allow required fields nested in an optional object. So if we passed in the following
```
{ "name": "account1" }
```
the request object `data` would come in as
```
{
  "name": "account1"
  "createdAt": {
    hash: undefined
    sequence: undefined
  }
}
```

I could probably make this code a little cleaner but it would be more complex if we actually wanted to validate that if you pass in `createdAt` field that both `hash` and `sequence` are defined. This is one of the main problems with yup I've run into. It gets even more difficult if you want to make `createdAt` nullable (which it techincally should be since you can set `createdAt` to null in the wallet)

Reply:
I pulled the PR down and it works just fine for me, you just have the YUP schema defined incorrectly. For optional sub fields you need this syntax.

```
export const CreateAccountRequestSchema: yup.ObjectSchema<CreateAccountRequest> = yup
  .object({
    name: yup.string().defined(),
    default: yup.boolean().optional(),
    createdAt: yup
      .object({
        hash: yup.string().defined(),
        sequence: yup.number().defined(),
      })
      .notRequired()
      .default(undefined),
  })
  .defined()
```

Once you do this, you can delete the entire block of code setting the createdAt as undefined.